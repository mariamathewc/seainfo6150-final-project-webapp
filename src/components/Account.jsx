import React from 'react';
import styles from "./Account.module.css";
import { NavLink } from 'react-router-dom';
import { useLocation } from "react-router";

const Account = (props) => {

	const location = useLocation();

    const mail = "mailto:".concat( location.state.email );

    return (
        <div className={styles.title}>
        <div className={styles.row}>
            <div className={styles.column1}>
                <h2><em>Personal Information</em></h2>
                <br />
                <br />
                <p><strong>First Name:</strong> {location.state.firstname}</p>
                <br />
                <p><strong>Last Name:</strong> {location.state.lastname}</p>
                <br />
                <p><strong>Gender:</strong> {location.state.genders}</p>
                <br />
                <p><strong>Date of Birth:</strong> {location.state.birthday} {location.state.birthmonth} {location.state.birthyear}</p>

        </div>
        <div className={styles.column2}>
                <h2><em>Address Information</em></h2>
                <br />
                <br />
                <p><strong>Street Address:</strong> {location.state.address}</p>
                <br />
                <p><strong>City:</strong> {location.state.city}</p>
                <br />
                <p><strong>State:</strong> {location.state.state}</p>
                <br />
                <p><strong>Zip Code:</strong> {location.state.zipcode}</p>
                <br />
                <p><strong>Country:</strong> {location.state.country}</p>
            </div>

            <div className={styles.column3}>
                <h2><em>Contact Information</em></h2>
                <br />
                <br />
                <p><strong>Phone No:</strong> {location.state.phone}</p>
                    <br />
                    
                <p><strong>Email:</strong><a href={mail}>{location.state.email}</a></p>
                
            </div>

            </div>
</div>

	)

}

export default Account 