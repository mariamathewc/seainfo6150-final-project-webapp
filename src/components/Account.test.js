import React from "react";
import Account from "./Account.jsx";
import { render } from "@testing-library/react";
import routeData from 'react-router';

const mockLocation = {
	pathname: '/account',
	state: {
		firstname: "Maria",
		lastname: "Mathew",
		address: "1700 N 1st St",
		city: "San Jose",
		state: "California",
		zipcode: "95112",
		country: "USA",
		email: "mariamathewc@gmail.com",
		phone: "5103092054",
		genders: "Female",

		birthday: "06",
		birthmonth: "01",
		birthyear: "1992"

	}
}


describe("Account tests", () => {
	it("renders correctly", () => {
		const { container } = jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
		expect(container).toMatchSnapshot();
	});
});

