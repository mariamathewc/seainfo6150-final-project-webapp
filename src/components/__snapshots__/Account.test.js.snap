// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account test renders correctly 1`] = `undefined`;
